subsys/bluetooth/mesh/cfg_srv.c

```diff
@@ -3319,8 +3319,6 @@ static void mod_reset(struct bt_mesh_model *mod, struct bt_mesh_elem *elem,
 		if (clear_count) {
 			bt_mesh_store_mod_sub(mod);
 		}
-
-		bt_mesh_model_data_store(mod, vnd, NULL, 0);
```

Comment:
Add a note in the reset callback API?
For instance

`@note If the model stores any persistent data, this needs to be erased manually.`